packages/messages/src/browser/notifications-message-client.ts

```diff
@@ -35,14 +41,17 @@ export class NotificationsMessageClient extends MessageClient {
             label: action,
             fn: element => onCloseFn(action)
         });
+
+        const timeout = (message.timeout || 0);
```

Comment:
if `message.actions` are undefined or empty, timeout should be undefined, right?

Reply:
you're right! the existence of actions should be the first criteria.

but I think instead of patching the timeout in `NotificationsMessageClient.showMessage` and evaluating everything in `Notifications.contructor`, we should compute the effective timeout here onve. considering the requirement could we calculate the timeout like this?
```
const timeout = (message.action)
  ? undefined
  : (message.timeout || this.preferences['notification.timeout'])
```

could you update the PR, please?